l2metrics/average_performance.py

```diff
@@ -0,0 +1,127 @@
+"""
+Copyright © 2021-2022 The Johns Hopkins University Applied Physics Laboratory LLC
+
+Permission is hereby granted, free of charge, to any person obtaining a copy 
+of this software and associated documentation files (the “Software”), to 
+deal in the Software without restriction, including without limitation the 
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or 
+sell copies of the Software, and to permit persons to whom the Software is 
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in 
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, 
+WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR 
+IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+import logging
+
+import numpy as np
+import pandas as pd
+
+from ._localutil import fill_metrics_df
+from .core import Metric
+
+logger = logging.getLogger(__name__)
+
+
+class AvgPerf(Metric):
+    name = "Average Performance"
+    capability = "adapt_to_new_tasks"
+    requires = {"syllabus_type": "agent"}
+    description = (
+        "Calculates the average performance over all blocks,"
+        "effectively computing an area under the performance curve for training blocks"
```

Comment:
Minor comment: for training and evaluation blocks after the two were merged into this single file

Reply:
@nguyeeq1 I was wondering about that, but the area-under-the-curve is only true for training blocks. I could have a separate statement about eval blocks, but I'm not sure what to say

Reply:
Hmm fair enough since eval episodes are independent...I guess a thought could be to make this most generically a mean regime performance metric. So there wouldn't be a need to distinguish between train and eval here since the calculation applies to every regime